### PR TITLE
Add tests for db, gofr, grpc, handler and httpServer

### PR DESCRIPTION
**Pull request by vipul-rawat**

### CODE REVIEW

File: pkg/gofr/container/db.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package container
 import (
 	"context"
 	"database/sql"
-	"fmt"
+	"gofr.dev/pkg/gofr/logging"
```
Comment (aryanmehrotra) [inline pkg/gofr/container/db.go:6]: should be at last

File: pkg/gofr/httpServer_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	// Send a GET request to the server
+	resp, err := http.Get("http://localhost:8080")
+	if err != nil {
+		t.Errorf("Failed to send GET request: %v", err)
+	}
+
+	// Check if the server is listening
+	if resp.StatusCode != http.StatusOK {
+		t.Errorf("Expected status code %d, but got %d", http.StatusOK, resp.StatusCode)
```
Comment (aryanmehrotra) [inline pkg/gofr/httpServer_test.go:45]: assert can be used

File: pkg/gofr/handler_test.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
+package gofr
```
Comment (aryanmehrotra) [inline pkg/gofr/handler_test.go:1]: test desc should be added in all assert statements
"TEST[%d], Failed.\n%s", i, tc.desc

File: pkg/gofr/handler_test.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+import (
+	"context"
+	"errors"
+	"github.com/stretchr/testify/assert"
+	"gofr.dev/pkg/gofr/container"
+	"gofr.dev/pkg/gofr/http/response"
+	"gofr.dev/pkg/gofr/logging"
+	"net/http"
+	"net/http/httptest"
```
Comment (aryanmehrotra) [inline pkg/gofr/handler_test.go:11]: sort imports

**Review by aryanmehrotra — CHANGES_REQUESTED** [review]

File: pkg/gofr/gofr.go [hunk 5]
```diff
@@ -39,12 +39,12 @@ type App struct {
 
 // RegisterService adds a grpc service to the gofr application.
 func (a *App) RegisterService(desc *grpc.ServiceDesc, impl interface{}) {
-	a.grpcRegistered = true
 	a.container.Logger.Infof("Registering GRPC Server: %s", desc.ServiceName)
 	a.grpcServer.server.RegisterService(desc, impl)
+	a.grpcRegistered = true
```
Comment (aryanmehrotra) [inline pkg/gofr/gofr.go:44]: why this change?

File: pkg/gofr/gofr_test.go [hunk 6]
```diff
@@ -14,3 +19,79 @@ func TestNewCMD(t *testing.T) {
...
+
+		if res.Data != tc.response {
+			t.Errorf("Unexpected response for %s %s. \t expected: %s \t got: %s", tc.method, tc.target, tc.response, res.Data)
+		}
+
+		if ctype := w.Header().Get(tc.headerKey); ctype != tc.headerVal {
+			t.Errorf("Header mismatch for %s %s. \t expected: %s \t got: %s", tc.method, tc.target, tc.headerVal, ctype)
+		}
+	}
+}
```
Comment (aryanmehrotra) [inline pkg/gofr/gofr_test.go:93]: assert?

**Review by aryanmehrotra — CHANGES_REQUESTED** [review]

File: pkg/gofr/gofr.go (hunk 5 above)
Comment (vipul-rawat) [reply pkg/gofr/gofr.go:44]: was suggested, since we should set it to be true after it is actually registered!

**Review by aryanmehrotra — APPROVED** [review]
